bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/api/WebexTeamsApi.java

```diff
@@ -0,0 +1,178 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.webexteams.internal.api;
+
+import static org.openhab.binding.webexteams.internal.WebexTeamsBindingConstants.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.util.InputStreamResponseListener;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.webexteams.internal.WebexAuthenticationException;
+import org.openhab.core.auth.client.oauth2.AccessTokenResponse;
+import org.openhab.core.auth.client.oauth2.OAuthClientService;
+import org.openhab.core.auth.client.oauth2.OAuthException;
+import org.openhab.core.auth.client.oauth2.OAuthResponseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * WebexTeamsApi implements API integration with Webex Teams.
+ * 
+ * Not using webex-java-sdk since it's not in a public maven repo, and it doesn't easily
+ * support caching refresh tokens between openhab restarts, etc..
+ * 
+ * @author Tom Deckers - Initial contribution
+ * 
+ */
+@NonNullByDefault
+public class WebexTeamsApi {
+
+    private final Logger logger = LoggerFactory.getLogger(WebexTeamsApi.class);
+
+    private final @NonNullByDefault({}) OAuthClientService authService;
+    private final @NonNullByDefault({}) HttpClient httpClient;
+
+    /**
+     * Constructor.
+     * 
+     * @param WebexTeamsHandler hander interact with handler context.
+     */
+    public WebexTeamsApi(OAuthClientService authService, HttpClient httpClient) {
+        this.authService = authService;
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Get a <code>Person</code> object for the account.
+     * 
+     * @return a <code>Person</code> object
+     * @throws WebexAuthenticationException when authentication fails
+     * @throws WebexTeamsApiException for other failures
+     */
+    public Person getPerson() throws WebexTeamsApiException, WebexAuthenticationException {
+        URI url = getUrl(WEBEX_API_ENDPOINT + "/people/me");
+
+        Person person = request(url, HttpMethod.GET, Person.class, null);
+        return person;
+    }
+
+    private URI getUrl(@Nullable String url) throws WebexTeamsApiException {
```

Comment:
The parameter will never be null so you can remove @Nullable

Maybe getUri would be a better name ?

Reply:
Removed @nullable and changed name.